### PR TITLE
Feature - PD report output detail #34

### PR DESCRIPTION
**Pull request by n1207n**
##### Feature list
* Created IndicatorReport Detail API for PD report output detail page to use

* _Describe the list of features from Polymer_

##### Progress checker
- [x] Django

- [ ] Polymer

- [x] Django test cases

- [ ] Polymer test cases

### CODE REVIEW

File: django_api/django_api/apps/indicator/tests/test_views.py [hunk 1]
```diff
@@ -132,3 +132,24 @@ def test_list_api(self):
         self.assertEquals(response.status_code, status.HTTP_200_OK)
         self.assertEquals(len(response.data), 1)
         self.assertNotEquals(response.data[0]['indicator_location_data'][0]['disaggregation'], {})
+
+
+class TestIndicatorReportDetailAPIView(APITestCase):
```
Comment (lkolacz) [inline django_api/django_api/apps/indicator/tests/test_views.py]: You should use `BaseAPITestCase` from core.tests.base

File: django_api/django_api/apps/indicator/tests/test_views.py [hunk 2]
```diff
@@ -132,3 +132,24 @@ def test_list_api(self):
...
+
+    def setUp(self):
```
Comment (lkolacz) [inline django_api/django_api/apps/indicator/tests/test_views.py]: then You can rm this fun .. 
if you need 5 quantity... just set class attr
`    generate_fake_data_quantity = 5`

File: django_api/django_api/apps/indicator/tests/test_views.py (hunk 1 above)
Comment (n1207n) [inline django_api/django_api/apps/indicator/tests/test_views.py]: Now that we got `BaseAPITestCase` in the codebase, I'll use this now.

File: django_api/django_api/apps/indicator/tests/test_views.py (hunk 2 above)
Comment (n1207n) [inline django_api/django_api/apps/indicator/tests/test_views.py]: `generate_fake_data_quantity` is quite handy, but it creates unnecessary data to this API as well. We may want to improve this attribute later in the future.

For now, I'll use this one.

File: django_api/django_api/apps/indicator/tests/test_views.py [hunk 3]
```diff
@@ -90,3 +90,22 @@ def test_list_api(self):
...
+    generate_fake_data_quantity = 5
+
+    def test_list_api(self):
+        self.reportable = Reportable.objects.filter(
+            lower_level_outputs__reportables__isnull=False
+        ).last()
+
+        self.indicator_report = self.reportable.indicator_reports.last()
+
+        url = reverse('indicator-report-detail-api', kwargs={'pk': self.reportable.id, 'indicator_report_pk': self.indicator_report.id})
```
Comment (lkolacz) [inline django_api/django_api/apps/indicator/tests/test_views.py]: PEP8 - max length of line

File: django_api/django_api/apps/indicator/urls.py [hunk 4]
```diff
@@ -12,5 +13,6 @@
         PDReportsAPIView.as_view(),
         name="programme-document-reports"),
     url(r'^$', IndicatorListCreateAPIView.as_view(), name='indicator-list-create-api'),
+    url(r'^(?P<pk>\d+)/indicator-reports/(?P<indicator_report_pk>\d+)$', IndicatorReportDetailAPIView.as_view(), name='indicator-report-detail-api'),
```
Comment (lkolacz) [inline django_api/django_api/apps/indicator/urls.py]: PEP8 - max length of line

File: django_api/django_api/apps/indicator/views.py [hunk 5]
```diff
@@ -128,3 +133,25 @@ def get(self, request, pk, format='json'):
...
+    """
+
+    def get_object(self, pk, indicator_report_pk):
+        reportable = get_object_or_404(Reportable, pk=pk)
+
+        try:
+            indicator_report = reportable.indicator_reports.get(id=indicator_report_pk)
+
+            return indicator_report
+        except Exception as e:
```
Comment (lkolacz) [inline django_api/django_api/apps/indicator/views.py]: Why You did't use Model.DoesNotExist exception definition?

File: django_api/django_api/apps/indicator/serializers.py [hunk 6]
```diff
@@ -62,6 +62,22 @@ class Meta:
         )
 
 
+class IndicatorReportDetailSerializer(serializers.ModelSerializer):
+    indicator_location_data = SimpleIndicatorLocationDataListSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = IndicatorReport
+        fields = (
+            'id', 'title',
```
Comment (lkolacz) [inline django_api/django_api/apps/indicator/serializers.py]: split it to two lines please

**Comment by chachra** [comment]
Please assess status / code in this PR and close it out @n1207n

**Comment by n1207n** [comment]
`develop` branch has all relevant features from this branch for IndicatorReport already and the only differentiator is IndicatorReport detail API. However, we don't query IndicatorReport individually because it is always list operation.

There I'm going to resolve this issue and close.

**Pull request closed without merging**